### PR TITLE
convert RMSNorm to NNX

### PR DESCRIPTION
**Pull request by cgarciae**
# Description

### CODE REVIEW

**Review by bvandermoon — COMMENTED** [review]
Do you know why the full test suite didn't run for the checks?

File: MaxText/layers/normalizations.py [hunk 1]
```diff
@@ -17,35 +17,68 @@
...
   """RMS normalization."""
 
-  epsilon: float = 1e-6
-  dtype: Any = jnp.float32
-  weight_dtype: Any = jnp.float32
-  kernel_axes: Tuple[Optional[str], ...] = ()
-  scale_init: Initializer = nn.initializers.ones
+  def __init__(
+      self,
+      features: int,
```
Comment (bvandermoon) [inline MaxText/layers/normalizations.py]: Is it possible to make this more descriptive like in https://github.com/AI-Hypercomputer/maxtext/pull/1840?

Comment (cgarciae) [reply MaxText/layers/normalizations.py]: Changing it to `num_features`.

Comment (bvandermoon) [reply MaxText/layers/normalizations.py]: Thanks @cgarciae. As a nit, should it be `features_shape` to match the style from #1840? Not 100% sure though since this is strictly an int

Comment (cgarciae) [reply MaxText/layers/normalizations.py]: In this case its not a shape (tuple). I'm also not super convinced about the shape semantics in `DenseGeneral` because you can also pass and `int`.

Comment (bvandermoon) [reply MaxText/layers/normalizations.py]: Agreed. Let me know if you have any name suggestions. Happy to update if we have something better

File: MaxText/layers/initializers.py [hunk 2]
```diff
@@ -39,3 +40,12 @@ def init_fn(key, shape, dtype, in_axis, out_axis):
     return fn(key, shape, dtype)
 
   return init_fn
+
+def variable_to_logically_partitioned(variable: nnx.VariableState):
```
Comment (cgarciae) [inline MaxText/layers/initializers.py:44]: @bvandermoon I'm thinking of moving this function to `initializers.py` since its shared but maybe there is a better place?

Comment (bvandermoon) [reply MaxText/layers/initializers.py:44]: I'm ok with it here. Could also make a separate file like `variable_partitions.py` or something which would probably be better IMO

File: MaxText/layers/normalizations.py (hunk 1 above)
Comment (bvandermoon) [reply MaxText/layers/normalizations.py]: Quick follow up on this - think this update may have been missed. Good to see the tests are now passing though

File: MaxText/layers/models.py [hunk 3]
```diff
@@ -380,7 +381,7 @@ def get_norm_layer(self):
         DecoderBlockType.SIMPLE_MLP,
         DecoderBlockType.LLAMA4,
     ):
-      return RMSNorm
+      return functools.partial(rms_norm, num_features=num_features)
```
Comment (bvandermoon) [inline MaxText/layers/models.py:389]: Why does this need to be changed to functools.partial?

**Review by bvandermoon — APPROVED** [review]

**Review by richjames0 — APPROVED** [review]
lgtm
